ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -92,26 +98,53 @@ private[server] object ServerHelpers {
           case Right(()) => Sync[F].pure(())
         }
 
+    def postProcessResponse(req: Request[F], resp: Response[F]): F[Response[F]] = {
+      val reqHasClose = req.headers.get(Connection).exists(_.hasClose)
+      val respConnection = resp.headers.get(Connection)
+      val connection: Connection =
+        if (reqHasClose) Connection(NonEmptyList.of("close".ci))
```

Comment:
This, and `keep-alive`, seem like good candidates for constant headers.  Headers like these with low cardinality, or a common value, are an area that we could better optimize than we do today.